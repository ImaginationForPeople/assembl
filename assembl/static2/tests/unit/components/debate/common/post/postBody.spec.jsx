import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import PostBody from '../../../../../../js/app/components/debate/common/post/postBody';

describe('PostBody component', () => {
  it('should render a post body', () => {
    const bodyDivRefSpy = jest.fn();
    const props = {
      body: '<p>You can\'t index the port without programming the wireless HTTP program!</p>',
      bodyDivRef: bodyDivRefSpy,
      bodyMimeType: 'text/*',
      contentLocale: 'fr',
      id: 'XYZ333',
      lang: 'fr',
      subject: <span>open-source Associate</span>,
      originalLocale: 'en',
      translate: true,
      translationEnabled: true
    };
    const renderer = new ShallowRenderer();
    renderer.render(<PostBody {...props} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});