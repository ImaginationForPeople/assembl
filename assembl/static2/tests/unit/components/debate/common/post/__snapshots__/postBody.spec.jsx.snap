// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostBody component should render a post body 1`] = `
<div
  className="post-body"
>
  <Apollo(Connect(Component))
    contentLocale="fr"
    id="XYZ333"
    lang="fr"
    originalLocale="en"
    translate={true}
  />
  <h3
    className="dark-title-3"
  >
    <span>
      open-source Associate
    </span>
  </h3>
  <div
    className="body "
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>You can't index the port without programming the wireless HTTP program!</p>",
      }
    }
  />
</div>
`;
